qualtran/bloqs/data_loading/qroam_clean_test.py

```diff
@@ -0,0 +1,141 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import numpy as np
+import pytest
+import sympy
+
+from qualtran.bloqs.data_loading.qroam_clean import (
+    _qroam_clean_multi_data,
+    _qroam_clean_multi_dim,
+    QROAMClean,
+    QROAMCleanAdjoint,
+)
+from qualtran.symbolics import ceil
+
+
+def test_bloq_examples(bloq_autotester):
+    bloq_autotester(_qroam_clean_multi_data)
+    bloq_autotester(_qroam_clean_multi_dim)
+
+
+def test_t_complexity_1d_data_symbolic():
```

Comment:
Is it worth adding a test which uses QROAM then QROAM^ in its decomposition to make sure the allocated RIGHT registers match the LEFTs which will be deallocated in the inverse?

Reply:
The classical simulation tests assert this; and also the fact that adjoint has the right lookup table to load.

Reply:
Got it. Nice.